### PR TITLE
feat: upgrade wallet connect to v3

### PR DESCRIPTION
**Pull request by mohandast52**
- Upgraded WalletConnect to version 3 (latest)
- Updated dependent packages including ethers.js and a few others

https://github.com/valory-xyz/autonolas-member-frontend/assets/22061815/354d88bc-6539-47ab-a7bf-b24acc7f6f75

checks and shows modal

### CODE REVIEW

**Review by truemiller — DISMISSED** [review]
Looks great in the video, code looks good!

**Comment by truemiller** [comment]
Small note: I get this when trying to preview on Vercel, unsure what's causing serverless function to timeout, likely a Vercel issue..
image

**Comment by oaksprout** [comment]
@mohandast52 I'm getting the same 504 as @truemiller so haven't been able to test properly.

**Review by truemiller — COMMENTED** [review]
seems good, have added 45s timeout as "/" serverless function was running for around 20s before responding

**Review by truemiller — DISMISSED** [review]

**Review by oaksprout — DISMISSED** [review]
lgtm – I was able to trigger a lock txn.

**Review by Tanya-atatakai — DISMISSED** [review]
Looks good to me, left a few comments

**Review by Tanya-atatakai — APPROVED** [review]

File: common-util/functions/index.jsx [hunk 1]
```diff
@@ -87,25 +87,25 @@ export const sendTransaction = (fn, account) => sendTransactionFn(fn, account, {
  */
 export const formatToEth = (value, dv = 0) => {
   if (isNil(value)) return dv || 0;
-  return (+ethers.utils.formatEther(value)).toFixed(2);
+  return (+ethers.formatEther(value)).toFixed(2);
```
Comment (truemiller) [inline common-util/functions/index.jsx:90]: while this is cool, I don't think the `+` is a commonly used for turning strings into numbers, lets keep it readable for others
```suggestion
  return Number(ethers.formatEther(value)).toFixed(2);
```

**Review by truemiller — APPROVED** [review]

**Pull request merged**